scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,249 @@
+        return azul_hit
+
+    files = list(map(change_datafile_uuid, hits))
+
+    # A bit more data massage to return only the file metadata (Easier parsing for later)
+    files = [hit['files'][0] for hit in files]
+
+    return files
+
+
+def parse_azul_metadata(azul_file_list: list) -> dict:
+    """
+    Parse the azul metadata about the files and return a map {uuid: metadata}, excluding DCP1/2 analysis files
+    :param azul_file_list:          File list from azul, output from get_files_metadata_from_azul()
+    :return file_metadata_by_uuid:  Dictionary in the format {file_uuid:file_metadata} for all files in azul_file_list
+    """
+    uuid_metadata_map = {file['uuid']: file for file in azul_file_list}
+
+    filesource_invalid_values = ["HCA Release", "DCP/2 Analysis", "DCP/1 Matrix Service"]
+    uuid_metadata_map_filtered = {uuid: metadata for uuid, metadata in uuid_metadata_map.items()
+                                  if not any([source == metadata['fileSource'] for source in filesource_invalid_values])}
+
+    return uuid_metadata_map_filtered
+
+
+def set_up_ingestapi(ingest_token: str, ingest_base_url: str) -> IngestApi:
+    """
+    Set up ingestApi object, with credentials to access and modify information in the database. URL is not asked because
+    this script can only be run in production (Azul service not available in dev/staging)
+
+    :param ingest_token:    Token to access/modify entities in ingest
+    :param ingest_base_url: Ingest's API base URL. Defaults to prod
+    :return ingest_api:     IngestApi object set up with token
+    """
+    ingest_api = IngestApi(ingest_base_url)
+    ingest_api.set_token(f"Bearer {ingest_token}")
+    return ingest_api
+
+
+def patch_file_metadata(uuid_metadata_map: dict, ingest_api: IngestApi, dry_run: bool) -> [list, list]:
+    """
+    Send a PUT request to each of the files in the uuid_size_map with the updated size and fileContentType extracted
+    from Azul
+    :param uuid_metadata_map:   {file_uuid: metadata} map of all the files that need to be updated
+    :param ingest_api:          IngestApi object with headers set up
+    :param dry_run:             bool value, if True no PUT operations will be performed (But information will be logged)
+    :return not_found:          List of file metadata that was not found in ingest
+    :return already_filled:     List of file metadata that was already filled
+    """
+    not_found = []
+    already_filled = []
+    for uuid, metadata in tqdm.tqdm(uuid_metadata_map.items()):
+        print(uuid, metadata)
+        # Check file exists in Ingest (Non-organically described matrices are virtually indistinguishable via metadata)
+        try:
+            logging.info(f"Searching for file metadata for file {metadata['name']}")
+            entity = ingest_api.get_entity_by_uuid('files', uuid)
+        except HTTPError:
+            not_found.append(metadata)
+            continue
```

Comment:
I would extract this to a function called `check_file_exists_in_ingest`.
If you do this to all the blocks in the `patch_file_metadata`, your code would become more readable, and will only show relevant details. You will have a series of high level, domain centric functions like `check_file_exists_in_ingest` and `check_entity_is_not_filled_in`, so the reader can understand the logic, with out being bombarded with implementation details like iterating on lists, api calls and logging.